### PR TITLE
Create ruff.yml

### PR DESCRIPTION
**Pull request by waltsims**
Related to and closes [#132](https://github.com/waltsims/k-wave-python/issues/132)

Adds code quality check with Ruff.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -72,3 +72,16 @@ exclude = [
 
 [tools.pytest.ini_options]
 testpaths = ["tests"]
+
+[tool.ruff]
+# Allow lines to be as long as 140 characters.
+line-length = 140
```
Comment (faridyagubbayli) [inline pyproject.toml:78]: Manually solved ~90 errors to set the line length to 140. Going down to 120 or 100 requires solving 103 or 853 more errors, respectively. I propose we keep this at 140 for now and update it once we have an automatic code formatting tool. We can still write all new code according to 100 chars rule. @waltsims wdyt?

File: kwave/executor.py
Comment (waltsims) [inline kwave/executor.py:1]: This gets removed in #174

File: kwave/utils/atten_comp.py [hunk 2]
```diff
@@ -294,7 +286,7 @@ def findClosest(arr, value):
         tv_filter[t_index, :] = np.roll(tv_filter[t_index, :], t_index)
 
     # zero out lower and upper triangles
-    ones_mat = np.ones((N, N))
+    np.ones((N, N))
```
Comment (waltsims) [inline kwave/utils/atten_comp.py:289]: Can this line be omitted?

File: kwave/utils/io.py [hunk 3]
```diff
@@ -277,7 +277,7 @@ def write_attributes(filename: str, file_description: Optional[str] = None, lega
         attributes = {
             h5_literals.FILE_MAJOR_VER_ATT_NAME: h5_literals.HDF_FILE_MAJOR_VERSION,
             h5_literals.FILE_MINOR_VER_ATT_NAME: h5_literals.HDF_FILE_MINOR_VERSION,
-            h5_literals.CREATED_BY_ATT_NAME: f'k-Wave N/A',
+            h5_literals.CREATED_BY_ATT_NAME: 'k-Wave N/A',
```
Comment (waltsims) [inline kwave/utils/io.py:280]: `k-wave-python __version__`?

File: pyproject.toml (hunk 1 above)
Comment (waltsims) [reply pyproject.toml:78]: Sounds good. I've added it to the project.

**Review by waltsims — COMMENTED** [review]
looks good!

File: kwave/utils/atten_comp.py (hunk 2 above)
Comment (faridyagubbayli) [reply kwave/utils/atten_comp.py]: Yeap, removed it.

File: kwave/utils/io.py (hunk 3 above)
Comment (faridyagubbayli) [reply kwave/utils/io.py]: Updated to use `kwave.VERSION`, good point!

**Review by faridyagubbayli — APPROVED** [review]

**Pull request merged**